ntoskrnl/po/power.c

```diff
@@ -124,12 +124,16 @@ PopPresentIrp(
     return Status;
 }
 
+static IO_COMPLETION_ROUTINE PopRequestPowerIrpCompletion;
+
+_Use_decl_annotations_
 static
 NTSTATUS
 NTAPI
-PopRequestPowerIrpCompletion(IN PDEVICE_OBJECT DeviceObject,
-                             IN PIRP Irp,
-                             IN PVOID Context)
+PopRequestPowerIrpCompletion(
+    PDEVICE_OBJECT DeviceObject,
+    PIRP Irp,
+    PVOID Context)
```

Comment:
`PoRequestPowerIrp` function got its SAL annotations improved below, however this function just lost its SAL annotations. :eyes:

Reply:
I added a new annotation to it. This is because the completion routine is a very common prototype.

Reply:
Annotations are also for humans to understand so that they can use functions in their intended way, and not just the automated code verifier. Now, the code verifier may know what `_Use_decl_annotations_` means, but as a human programmer I don't know anymore whether DeviceObject is IN or OUT, Irp is IN or OUT, and Context is IN or OUT (or something else).

Reply:
Unfortunately you can't cherry-pick annotations, so the trade-off is this:
* either use `_Use_decl_annotations` and don't repeat yourself between declaration and definition
* or always repeat the entirety of the annotations, even if they're ridiculously extensive.

Unless/until we agree on a set of rules (which _could_ say to sometimes do one, sometimes the other), this discussion is about everyone's personal preferences and therefore not useful.